node_modules/oae-authz/lib/permissions.js

```diff
@@ -525,6 +525,26 @@ var _validateRoleChanges = function(ctx, resource, targetRoles, opts, callback)
     var principalTargetRoles = _.first(targetRoles);
     var emailTargetRoles = _.last(targetRoles);
 
+    // Ensure that no guests can get invited on a tenant that has disabled
+    // inviting guests. When a tenant disables inviting guests, email
+    // addresses that match known tenant email domains should still get an
+    // invitation. Email addresses that wind up on the guest tenant should
+    // be disallowed however.
+    var guestEmails = !_.chain(emailTargetRoles)
+        .pluck('email')
+        .map(TenantsAPI.getTenantByEmail)
+        .filter({'isGuestTenant': true})
+        .value();
+
+    if (!_.isEmpty(guestEmails) && !TenantsUtil.canInviteGuests(ctx.user().tenant.alias)) {
+        var err = {
+            'code': 401,
+            'invalidPrincipals': guestEmails,
+            'msg': 'Guests cannot be invited from this tenant'
+        };
+        return callback(err);
```

Comment:
We don't need to assign a variable to `err` here, we can just instantiate the object in the callback parameter. Also avoids the hazard where `err` becomes a reference in the parent function scope.